File: Practise Random/RevVowel.py
class RevVowel:
    def __init__(self):
        pass
    def ReverseVowels(self,s):

        v= "aeiou"
        s= list(s.lower())
        f=0
        l=len(s)-1
        while f<l:
            if s[f] in v and s[l] in v:
                s[f],s[l]=s[l],s[f]
                f+=1
                l-=1
            elif s[f] not in v:
                f+=1
            elif s[l] not in v:
                l-=1
            else:
                f+=1
                l+=1
        return s

if __name__=="__main__":
    input_str=input("Enter String Name")
    c_o= RevVowel()
    reversed_v= c_o.ReverseVowels(input_str)
    print("The Reversed Vowel String is:",reversed_v)

File: Practise Random/RevVowel.cpp
#include <iostream>
#include <string>
#include<unordered_set>

using namespace std;

string reverseVowels(string s){
    unordered_set<char> vowels={ 'a','e','i','o','u','A','E','I','O','U' };
    int left =0;
    int right= s.length()-1;

    while(left<right){
        if(vowels.find(s[left])!=vowels.end() && vowels.find(s[right])!=vowels.end()){
            swap(s[left],s[right]);
            left++;
            right--;
        }else if(vowels.find(s[left])==vowels.end()){
            left++;
        }else if(vowels.find(s[right])==vowels.end()){
            right--;
        }else{
            left++;
            right--;
        }

    }
    return s;

}


int main()
{
    string input_string;
    cout<<"Enter the string length";
    getline(cin,input_string);

    string reversed_v= reverseVowels(input_string);
    cout<<"String with reversed vowels is"<<reversed_v;

    return 0;

}